tools/ossf_best_practices/passing_criteria.md

```diff
@@ -0,0 +1,397 @@
+
+**Met** Your email will be acknowledged within 24 hours
+
+
+# Quality
+
+## Working build system
+
+> If the software produced by the project requires building for use, the project MUST provide a working build system that can automatically rebuild the software from source code.
+
+**Met**
+https://github.com/nodejs/node#build
+
+
+> It is SUGGESTED that common tools be used for building the software.
+
+**Met**
+python, make or batch file, gcc or clang
+
+
+> The project SHOULD be buildable using only FLOSS tools.
+
+**Met**
+python, make or batch file, gcc or clang
+
+
+## Automated test suite
+
+> The project MUST use at least one automated test suite that is publicly released as FLOSS (this test suite may be maintained as a separate FLOSS project). The project MUST clearly show or document how to run the test suite(s) (e.g., via a continuous integration (CI) script or via documentation in files such as BUILD.md, README.md, or CONTRIBUTING.md).
+
+**Met**
+
+
+> A test suite SHOULD be invocable in a standard way for that language.
+
+**Met**
+make or batch file, executed using python
+
+
+> It is SUGGESTED that the test suite cover most (or ideally all) the code branches, input fields, and functionality.
+
+**Met**
+
+
+> It is SUGGESTED that the project implement continuous integration (where new or changed code is frequently integrated into a central code repository and automated tests are run on the result).
+
+**Met**
+https://ci.nodejs.org/
+
+
+## New functionality testing
+
+> The project MUST have a general policy (formal or not) that as major new functionality is added to the software produced by the project, tests of that functionality should be added to an automated test suite.
+
+**Met** 
+https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-5-test
+
+> The project MUST have evidence that the test_policy for adding tests has been adhered to in the most recent major changes to the software produced by the project.
+
+**Met**
+
+> It is SUGGESTED that this policy on adding tests (see test_policy) be documented in the instructions for change proposals
+
+**Met**
+
+## Warning flags
+
+> The project MUST enable one or more compiler warning flags, a "safe" language mode, or use a separate "linter" tool to look for code quality errors or common simple mistakes, if there is at least one FLOSS tool that can implement this criterion in the selected language.
+
+**Met**
+-Wall for compiling, c++ and JS linting as part of test suite
+
+
+> The project MUST address warnings.
+
+**Met**
+
+> It is SUGGESTED that projects be maximally strict with warnings in the software produced by the project, where practical.
+
+**Met**
+
+# Security
+
+## Secure development knowledge
+
+> The project MUST have at least one primary developer who knows how to design secure software. (See ‘details’ for the exact requirements.)
+
+**Met**
```

Comment:
I think this one and the next one is worth some discussion to undersatnd better. Who would we say is that person?

Reply:
Okay, that's a funny one. From the details:

> This requires understanding the following design principles, including the 8 principles from [Saltzer and Schroeder](https://web.mit.edu/Saltzer/www/publications/protection/)

Bear in mind that the referenced paper is from 1975. The 8 basic principles still apply for the most part, of course, but even back then, they were not considered authoritative or complete by the authors themselves. The only post-1975 extension is this as far as I can tell:

> * limited attack surface (the attack surface - the set of the different points where an attacker can try to enter or extract data - should be limited)
> * input validation with allowlists (inputs should typically be checked to determine if they are valid before they are accepted; this validation should use allowlists (which only accept known-good values), not denylists (which attempt to list known-bad values)).

Since the document only requires us to understand and not to actually follow or implement any of the principles, there should be plenty of qualified developers.

Reply:
I like your approach @tniessen. Based on that we can asume that we meet the criteria. But we will need to ensure that we can list that group of people or at least one, in order to do that we can do a separate discussion in https://github.com/nodejs/security-wg/issues/987